src/core/Camera.js

```diff
@@ -679,7 +685,16 @@ Phaser.Camera.prototype = {
         if (vx <= this.bounds.x * this.scale.x)
         {
             this.atLimit.x = true;
-            this.view.x = this.bounds.x * this.scale.x;
+
+            // if smoothBounds enabled, interpolate to new view position (at rate this.lerp.x) instead of jumping.
+            if (this.smoothBounds)
+            {
+                this.view.x = this.game.math.linear(this.view.x, this.bounds.x * this.scale.x, this.lerp.x);
```

Comment:
Yeah, I see what you mean.  I think what I'll do is add an internal `smoothBoundsTarget` object with x and y, and then just have the camera interpolate towards that during `update()`.  Then within `checkBounds()`, if `smoothBounds` is enabled, I'll just set the `smoothBoundsTarget` rather than "jumping" the view